### PR TITLE
Fixed ticket #924 (Problem with GeneralizedParetoFactory)

### PR DESCRIPTION
**Pull request by regislebrun**
The problem was in the method of exponential regression, where the case of a repeated minimum value in the sample was not correctly managed.

### CODE REVIEW

File: ChangeLog [hunk 1]
```diff
@@ -32,7 +32,7 @@
 === Miscellaneous ===
 
 === Bug fixes ===
-
+ * #924 (Probleme with the factory of a Generalized Pareto distribution)
```
Comment (jschueller) [inline ChangeLog]: could you move this to the 1.10 section ? It will be easier for me to backport to the 1.10 branch ?

**Comment by jschueller** [comment]
Side note: the doc adertises the classical method of moments:
"The classical moments method is proposed but not used by default."
but the buildMethodOfMoments method is not available.
Maybe it should be either removed, available with an argument, resourcemap key...

File: ChangeLog (hunk 1 above)
Comment (regislebrun) [reply ChangeLog]: @jschueller You mean, the 1.10rc1 section? Or do I have to create a new 1.10 section? There are currently only two sections: 1.11 and 1.10rc1.

Comment (jschueller) [reply ChangeLog]: Yes, 1.10rc1

File: lib/src/Uncertainty/Distribution/GeneralizedParetoFactory.cxx [hunk 2]
```diff
@@ -137,8 +149,13 @@ struct GeneralizedParetoFactoryParameterConstraint
...
-      if ((xNm1mJ == xMin) || (xNm2mJ == xMin)) sampleY_[j][0] = (j + 1.0) * SpecFunc::LogMaxScalar;
-      else sampleY_[j][0] = (j + 1.0) * std::log((xNm1mJ - xMin) / (xNm2mJ - xMin));
+      if ((xNm1mJ == xMin) || (xNm2mJ == xMin))
+	{
+	  const String message("Cannot use exponential regression to estimate a GeneralizedPareto distribution when the mininmum value of the sample is repeated");
+	  LOGINFO(message);
+	  throw InvalidArgumentException(HERE) << message;
+	}
+      sampleY_[j][0] = (j + 1.0) * std::log((xNm1mJ - xMin) / (xNm2mJ - xMin));
     }
```
Comment (sofianehaddad) [inline lib/src/Uncertainty/Distribution/GeneralizedParetoFactory.cxx]: sampleY_(j, 0)

**Comment by jschueller** [comment]
super!

**Pull request merged**